modules/cudafilters/test/test_filters.cpp

```diff
@@ -703,6 +704,18 @@ INSTANTIATE_TEST_CASE_P(CUDA_Filters, Median, testing::Combine(
     WHOLE_SUBMAT)
     );
 
-}} // namespace
+#ifdef __OPENCV_USE_WAVELET_MATRIX_FOR_MEDIAN_FILTER_CUDA__
```

Comment:
Lets run these tests regardless of optimization support.

Reply:
Thank you very much for your review.
For environments that are not supported by optimization support (e.g. CUDA 10 or lower), this test will always fail, is that still ok?

Reply:
In general API support should not depend on available optimizations. For such cases we usually have fallback generic code.

Do you mean that it fails on `CV_Assert(srcType == CV_8UC1);` assertion, right?

It makes sense to emit `Error::StsNotImplemented` error return instead of generic assertion.
I believe it is OK in CUDA cases to skip test if there is `Error::StsNotImplemented` error reaised.